From 8934bb63853f653acf7590833ccb7773a9e9c320 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9rgio=20M=2E=20Basto?= <sergio@serjux.com>
Date: Sun, 23 Nov 2025 22:16:33 +0000
Subject: [PATCH 3/9] Always call look_for_java to set JAVA_VERSION

This ensure that `--add-opens java.base/java.net=ALL-UNNAMED` is added to JAVA_ARGS for Java 11 or higher [1]

Previously, when JAVA_PROGRAM_DIR was set, look_for_java was not called,
and therefore JAVA_VERSION was not set.

[1]
    if echo "$JAVA_VERSION" | grep -E "^(1[1-9]|[2-9][0-9])\." > /dev/null ; then
      JAVA_ARGS="${JAVA_ARGS} --add-opens java.base/java.net=ALL-UNNAMED"
    fi
---
 core/src/com/biglybt/platform/unix/startupScript | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/core/src/com/biglybt/platform/unix/startupScript b/core/src/com/biglybt/platform/unix/startupScript
index ec457a173..a0dbfccd6 100644
--- a/core/src/com/biglybt/platform/unix/startupScript
+++ b/core/src/com/biglybt/platform/unix/startupScript
@@ -171,10 +171,8 @@ moveInSWT()
 echo "$MSG_STARTING"
 
 # locate and test the java executable
-if [ "$JAVA_PROGRAM_DIR" == "" ]; then
-	if ! look_for_java ; then
-		exit 1
-	fi
+if ! look_for_java ; then
+    exit 1
 fi
 
 # Java 11+ needs --add-opens, which borks some older java versions
-- 
2.51.1

